charts/otel-cloud-stack/values.yaml

```diff
@@ -0,0 +1,2141 @@
+  # # Environment variables for the target allocator
+  # env:
+  #   - name: ENV_VAR1
+  #     value: value1
+  #   - name: ENV_VAR2
+  #     value: value2
+  # # Observability configuration for the target allocator
+  # observability:
+  #   metrics:
+  #     enableMetrics: true
+
+  # Affinity configuration for the collector
+  affinity: {}
+  #   nodeAffinity:
+  #     requiredDuringSchedulingIgnoredDuringExecution:
+  #       nodeSelectorTerms:
+  #         - matchExpressions:
+  #             - key: kubernetes.io/e2e-az-name
+  #               operator: In
+  #               values:
+  #                 - e2e-az1
+  #                 - e2e-az2
+
+  # Lifecycle configuration for the collector
+  lifecycle: {}
+  #   preStop:
+  #     exec:
+  #       command:
+  #         [
+  #           "/bin/sh",
+  #           "-c",
+  #           "echo Hello from the preStop handler > /dev/termination-log",
+  #         ]
+
+  # Liveness probe configuration for the collector
+  livenessProbe: {}
+  #   initialDelaySeconds: 3
+  #   periodSeconds: 5
+  #   timeoutSeconds: 2
+  #   failureThreshold: 5
+
+  # Observability configuration for the collector
+  observability: {}
+  #   metrics:
+  #     enableMetrics: true
+
+  # Update strategy for the collector
+  updateStrategy: {}
+  #   type: RollingUpdate
+
+  # Volume mounts for the collector
+  volumeMounts: []
+  #   - name: data
+  #     mountPath: /data
+
+  # Ports configuration for the collector
+  # The operator automatically calculates ports for known receivers and exporters
+  # Set any custom ports here.
+  ports: []
+  # - name: http
+  #   protocol: TCP
+  #   port: 80
+  #   targetPort: 8080
+
+  # Environment variables for the collector
+  env: []
+  # - name: ENV_VAR1
+  #   value: value1
+  # - name: ENV_VAR2
+  #   value: value2
+
+  # Volume claim templates for the collector
+  volumeClaimTemplates: []
+  # - metadata:
+  #     name: storage
+  #   spec:
+  #     accessModes: ["ReadWriteOnce"]
+  #     resources:
+  #       requests:
+  #         storage: 1Gi
+
+  # Tolerations for the collector
+  tolerations: []
+  # - key: "key"
+  #   operator: "Equal"
+  #   value: "value"
+  #   effect: "NoSchedule"
+
+  # Volumes for the collector
+  volumes: []
+  # - name: config-volume
+  #   configMap:
+  #     name: config
+
+  # Init containers for the collector
+  initContainers: []
+  # - name: init-nginx
+  #   image: nginx
+
+  # Additional containers for the collector
+  additionalContainers: []
+  # - name: additional-container
+  #   image: busybox
+
+  # Topology spread constraints for the collector
+  topologySpreadConstraints: []
+  # - maxSkew: 1
+  #   topologyKey: kubernetes.io/hostname
+  #   whenUnsatisfiable: DoNotSchedule
+  #   labelSelector:
+  #     matchLabels:
+  #       app: my-app
+
+  # Config maps for the collector
+  configmaps: []
+  # - name: config
+  #   mountPath: /etc/config
+
+# Top level field specifying collectors configuration
+collectors:
```

Comment:
I'd prefer to nest all things related to the collector under 1 root section.

Reply:
the benefit of doing it this way is that anyone can add any collectors they want created to this key and they will be created as part of the chart. It also allows them to easily overlay values for specific collectors in different environments. i.e. if someone wanted to make a custom collector `xcol` and have a different sample rate in each env/region combo ([dev,prod] x [us-east-1, us-east-2]) they would be able to have a single xcol definition and only overlay the new sample rate rather than redefine the entire configuration.